muzzle/src/main/java/io/opentelemetry/javaagent/tooling/HelperInjector.java

```diff
@@ -127,13 +127,15 @@ public HelperInjector(
 
   public HelperInjector(
       String requestingName,
+      List<String> helperClassNames,
       Map<String, Function<ClassLoader, byte[]>> helperMap,
       List<HelperResource> helperResources,
       ClassLoader helpersSource,
       Instrumentation instrumentation) {
     this.requestingName = requestingName;
 
-    this.helperClassNames = helperMap.keySet();
+    this.helperClassNames = new LinkedHashSet<>(helperClassNames);
+    this.helperClassNames.addAll(helperMap.keySet());
     this.dynamicTypeMap.putAll(helperMap);
```

Comment:
Does this work? When you look at `getHelperMap()`, it either uses the `dynamicTypeMap` or `helperClassNames`, but never both at the same time.

Reply:
I think it probably worked because only one of `dynamicTypeMap` or `helperClassNames` was actually filled. I have reworked this to use 2 HelperInjectors.